doc/development/guide/architecture.rst

```diff
@@ -4,179 +4,278 @@
+to run the tapes on the device, and to post-process the results. The classical processing of this pipeline
+maintains differentiability, so that we can not only compute the result, but the gradient of the result
+with respect to the trainable parameters.
 
-Key design details
-##################
+|
 
-The following are key design details related to how PennyLane works internally.
+.. figure:: pl_workflow.png
+    :align: center
+    :width: 750px
 
-Operators
-*********
+|
 
-Quantum operators are incorporated by the :class:`~.Operator` class which
-contains basic information about the operator (e.g. number of parameters,
-number of wires it acts on, etc.) and further convenience methods (e.g.
-:attr:`~.Operator.matrix`, :attr:`~.Operator.eigvals`).
+The fact that multiple tapes can be constructed from one quantum function may be
+surprising at first, but there are many situations in which the evaluation of a quantum circuit
+practically requires many circuits to be evaluated, for example:
 
-Two important subclasses of the ``Operator`` class are:
+* When the observable is a Hamiltonian represented as a linear combination of Pauli words, the device may
+  instruct the QNode to create one circuit for each Pauli word, and to compute their linear combination
+  during post-processing.
+* When a gradient of the QNode is requested, and parameter-shift rules have to be used, the QNode
+  constructs tapes in which parameters are shifted, and recombines the result to return a gradient.
 
-* the :class:`~.Operation` class representing quantum gates,
-* the :class:`~.Observable` representing quantum observables specified for
-  measurement.
-
-Together with ``Operator``, these classes serve as base classes for quantum
-operators.
-
-Certain operators can serve as both quantum gates and observables (e.g.
-:class:`~.PauliZ`, :class:`~.PauliX`, etc.). Such classes inherit from both
-``Operation`` and ``Observable`` classes.
-
-Quantum operators are used to build quantum functions
-which are evaluated by a ``QNode`` on a bound device. Users can define such quantum
-functions by creating regular Python functions and instantiating ``Operator``
-instances in temporal order, one per line.
-
-The following is an example of this using the :func:`~pennylane.qnode` decorator and a
-valid pre-defined device (``dev``).
+Interfaces
+**********
 
-.. code-block:: python
+The construction of tapes, as well as post-processing the results from executing tapes,
+are classical computations, and they are "tracked" by the autodifferentiation framework.
+In other words, these steps can invoke differentiable classical computations, such as:
 
-    @qml.qnode(dev)
-    def circuit():
-        qml.PauliX(0)
-        return qml.expval(qml.PauliZ(0))
+* The decomposition of a user-defined gate into other gates that take some
+  function of the original gate's parameters
+* The linear re-combination of Hamiltonian terms with trainable coefficients.
```

Comment:
These sentences were a bit harder to parse compared to the rest of the article, which is very clearly written :)